### PR TITLE
Add support for Managed Grafana v10

### PR DESCRIPTION
**Pull request by bonclay7**
### What does this PR do?

- Adds support for v10 workspaces
- Introduces service account tokens for v9 and v10
- Updates documentation

### More

- [x] Yes, I ran `pre-commit run -a` with this PR
- [x] Yes, I added unit tests for new code added (as much as possible)

**Note**: Not all the PRs require unit tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,6 +55,30 @@ amazon-managed-grafana-migrator -v
...
 
+### Migrating to Amazon Managed Grafana v10
+
+v9 and v10 introduced Grafana Service Accounts which will be required by the
+migrator, especially for v10. Note that Service Accounts are billed as active
+users
+
+1. Creating a Service Account
+
+```console
```
Comment (lewinkedrs) [inline README.md:66]: Maybe you want to have an 'export' before this and grab the account id with jq. Theb just have this output be a bash environment variable. Then later on when the variable is used you can just say $SRC_SERVER_ACCOUNT_ID . That way people do not have to remember/scroll up and can just copy paste.

File: internal/pkg/app/app.go [hunk 2]
```diff
@@ -54,12 +54,5 @@ func (a *App) Run(srcCustomGrafanaClient CustomGrafanaClient) error {
 	log.Info()
```
Comment (lewinkedrs) [inline internal/pkg/app/app.go:54]: Is this empty log message just to create a newline? You could just start the next one with '\n' , and then remove this. That way each call to log actually has a corresponding message.

File: internal/pkg/aws/aws.go [hunk 3]
```diff
@@ -14,15 +14,36 @@ import (
 type api interface {
```
Comment (lewinkedrs) [inline internal/pkg/aws/aws.go:14]: I think it's a little confusing to call this package aws, especially because you are importing aws-sdk-go/aws . Can this package be named something else. maybe 'api' ? Also, can we not use the aws-sdk-go-v2 packages here? I suppose that is outside the scope of this specific pull request.

**Comment by bonclay7** [comment]
Thanks Kevin, all valuable feedback. I'll issue another PR for some 🧹
